File: backend/src/models/Maestro.js
// src/models/Maestro.js
import { Usuario } from './Usuario.js';

export class Maestro extends Usuario {
    constructor(idUsuario, password, name, email) {
        super(idUsuario, password); 
        this.name = name;
        this.email = email;
    }

    // Métodos administrativos nativos que propuso tu compañero para el futuro
    verAvanceIndividual(estudiante) {
        console.log(`Mostrando avance individual de: ${estudiante.name}`);
        return {}; // Stub para desarrollo posterior
    }

    verAvanceColectivo() {
        console.log("Mostrando avance colectivo del grupo.");
        return {}; // Stub para desarrollo posterior
    }

    filtrarInformacion(criterio) {
        console.log(`Filtrando estudiantes por criterio: ${criterio}`);
        return [];
    }
}

File: java-backend/src/main/java/com/example/mathexpedition/models/Maestro.java
package com.example.mathexpedition.models;

import java.util.ArrayList;
import java.util.List;

public class Maestro extends Usuario {
    private String name;
    private String email;

    public Maestro() {
        super();
    }

    public Maestro(String idUsuario, String name, String email, String password) {
        super(idUsuario, password);
        this.name = name;
        this.email = email;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Object verAvanceIndividual(Estudiante estudiante) {
        System.out.println("Mostrando avance individual de: " + estudiante.getName());
        return new Object();
    }

    public Object verAvanceColectivo() {
        System.out.println("Mostrando avance colectivo del grupo.");
        return new Object();
    }

    public List<Object> filtrarInformacion(String criterio) {
        System.out.println("Filtrando estudiantes por criterio: " + criterio);
        return new ArrayList<>();
    }
}
